packages/@expo/config-plugins/src/android/Updates.ts

```diff
@@ -65,15 +66,19 @@ const withUpdatesManifest: ConfigPlugin = (config) => {
 };
 
 const withRuntimeVersionResource = createStringsXmlPlugin(
-  applyRuntimeVersionFromConfig,
+  applyRuntimeVersionFromConfigAsync,
   'withRuntimeVersionResource'
 );
 
-export function applyRuntimeVersionFromConfig(
-  config: Pick<ExpoConfigUpdates, 'sdkVersion' | 'runtimeVersion'>,
+export async function applyRuntimeVersionFromConfigAsync(
+  config: Pick<ExpoConfig, 'sdkVersion' | 'runtimeVersion' | '_internal'>,
   stringsJSON: ResourceXML
-): ResourceXML {
-  const runtimeVersion = getRuntimeVersionNullable(config, 'android');
+): Promise<ResourceXML> {
+  const runtimeVersion = await getRuntimeVersionNullableAsync(
+    config._internal?.projectRoot,
```

Comment:
@Kudo I tried it out but typescript is complaining. 
First, it says that this property doesn't exist. So I have to put on the above Pick list.
Then, it says that `Type '"modRequest"' is not assignable to type 'keyof ExpoConfig'`.

Any ideas?

Reply:
this may help:

```patch
--- a/packages/@expo/config-plugins/src/android/Updates.ts
+++ b/packages/@expo/config-plugins/src/android/Updates.ts
@@ -11,7 +11,7 @@ import {
 } from './Manifest';
 import { buildResourceItem, ResourceXML } from './Resources';
 import { removeStringItem, setStringItem } from './Strings';
-import { ConfigPlugin } from '../Plugin.types';
+import { ConfigPlugin, ExportedConfigWithProps } from '../Plugin.types';
 import { createStringsXmlPlugin, withAndroidManifest } from '../plugins/android-plugins';
 import { withPlugins } from '../plugins/withPlugins';
 import {
@@ -70,9 +70,10 @@ const withRuntimeVersionResource = createStringsXmlPlugin(
 );

 export function applyRuntimeVersionFromConfig(
-  config: Pick<ExpoConfigUpdates, 'sdkVersion' | 'runtimeVersion'>,
+  config: ExportedConfigWithProps<Pick<ExpoConfigUpdates, 'sdkVersion' | 'runtimeVersion'>>,
   stringsJSON: ResourceXML
 ): ResourceXML {
+  const projectRoot = config.modRequest.projectRoot;
   const runtimeVersion = getRuntimeVersionNullable(config, 'android');
   if (runtimeVersion) {
     return setStringItem(
diff --git a/packages/@expo/config-plugins/src/plugins/android-plugins.ts b/packages/@expo/config-plugins/src/plugins/android-plugins.ts
index c5e479087f..ca7b566bd2 100644
--- a/packages/@expo/config-plugins/src/plugins/android-plugins.ts
+++ b/packages/@expo/config-plugins/src/plugins/android-plugins.ts
@@ -1,12 +1,10 @@
-import { ExpoConfig } from '@expo/config-types';
-
 import { withMod } from './withMod';
-import { ConfigPlugin, Mod } from '../Plugin.types';
+import { ConfigPlugin, Mod, ExportedConfigWithProps } from '../Plugin.types';
 import { Manifest, Paths, Properties, Resources } from '../android';

 type OptionalPromise<T> = T | Promise<T>;

-type MutateDataAction<T> = (expo: ExpoConfig, data: T) => OptionalPromise<T>;
+type MutateDataAction<T> = (expo: ExportedConfigWithProps, data: T) => OptionalPromise<T>;

 /**
  * Helper method for creating mods from existing config functions.
```

Reply:
@Kudo thanks! I've made slight changes to use the generics data. Can you take a look?